src/credential/trustmark/get-credential-trustmark.ts

```diff
@@ -0,0 +1,39 @@
+import { SignJWT, type CryptoContext } from "@pagopa/io-react-native-jwt";
+
+export type GetCredentialTrustmarkJwt = (
+  walletInstanceAttestation: string,
+  wiaCryptoContext: CryptoContext,
+  credentialType: string,
+  documentNumber?: string
+) => Promise<string>;
+
+/**
+ * Generates a trustmark signed JWT, which is used to verify the authenticity of a credential.
+ *
+ * @param walletInstanceAttestation the Wallet Instance's attestation
+ * @param wiaCryptoContext The Wallet Instance's crypto context associated with the walletInstanceAttestation parameter
+ * @param credentialType The type of credential for which the trustmark is generated
+ * @param documentNumber (Optional) Document number contained in the credential, if applicable
+ * @returns A promise that resolves to the signed JWT string, representing the credential's trustmark.
+ */
+export const getCredentialTrustmarkJwt: GetCredentialTrustmarkJwt = async (
+  walletInstanceAttestation,
+  wiaCryptoContext,
+  credentialType,
+  documentNumber
+): Promise<string> => {
+  const signedTrustmarkJwt = await new SignJWT(wiaCryptoContext)
+    .setProtectedHeader({
+      alg: "ES256",
+    })
+    .setPayload({
+      iss: walletInstanceAttestation,
+      sub: credentialType,
+      subtyp: documentNumber,
+    })
+    .setIssuedAt()
+    .setExpirationTime("2m")
+    .sign();
+
+  return signedTrustmarkJwt;
+};
```

Comment:
I suggest adding a runtime check to verify that the public key present in the walletInstanceAttestation is the one associated with wiaCryptoContext.

Reply:
Addressed here: https://github.com/pagopa/io-react-native-wallet/pull/156/commits/ae8c8c15c4b8ff1cc75c85fdcc377efb2b1f8c72